### PR TITLE
Migrate MDX doc components to Markdoc tags

### PR DESCRIPTION
**Pull request by nikomancy**
## Summary and Motivation
Migrating MDX documentation components into Markdoc to enable authoring content in Markdoc and migrating old MDX documentation over to Markdoc.

Here's a [video scrollthrough of the new components](https://www.loom.com/share/62691aa53b274fa3b3219b607f144cdc?sid=ceae6887-f9f6-4b5e-9ccc-81babd37ee4a).

The following components have been migrated

- Admonitions (Note and Warning)
- Buttons
- Crosses and Checks
- Images
- Status Badges
- Code Reference Links
- Reference Tables
- Article Lists

I've also added a markdown file into the root of the repository that documents each of these components that can be used as a convenient location for viewing and testing changes to our Markdoc setup.

### Testing Details
Frontend changes only. Tested on local environment at mobile, small, medium, and large breakpoints.

### CODE REVIEW

File: docs/markdoc-component-documentation.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
+---
```
Comment (PedramNavid) [inline docs/markdoc-component-documentation.md:1]: This is really neat, thanks for making this!

**Review by PedramNavid — APPROVED** [review]

File: docs/markdoc-component-documentation.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+You can invoke a cross with this tag `{% cross /}%` and it looks like this {% cross /%}. You can invoke checks with this tag `{% check /}%` and it looks like this {% check /%}.
+
+Crosses and checks can also be used in lists:
+- {% check /%} Completed task
+- {% cross /%} Incomplete task
+- {% cross /%} Super Incomplete task 
+
+You can also put crosses and checks into headers 
+
+### This is a header with a check {% check /%}
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:156]: I'm on the fence if this is worth addressing - and it doesn't have to be part of this PR, since it feels like scope creep - but the current logic for heading IDs/anchors appends an extra hyphen when the heading includes a component.

For example, this is the ID for this heading:

```
#this-is-a-header-with-a-check-
```

This is avoidable if we set the heading ID manually, but IMO it's probably easier to trim these hyphens at some point.

File: docs/markdoc-component-documentation.md [hunk 3]
```diff
@@ -0,0 +1,263 @@
...
+
+## Code Reference Links
+
+Code reference links let you point to a specific file in the codebase. They're useful for linking to examples or reference implementations.
+
+`{% codeReferenceLink filePath="examples/deploy_ecs" /%}`
+
+{% codeReferenceLink filePath="examples/deploy_ecs" /%}
+
+## Reference Tables
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:193]: Looking at this, I wonder if it's best to scrap this component and just focus on styling improvements + flexibility for regular ol' tables. Being able to use Markdown in tables like these is huge.

I made this component awhile back because I needed control that Markdown tables didn't provide (ex: setting custom widths, text alignment, etc) and I wanted to avoid a ton of table HTML. Notice how I've since avoided using it outside of a few pages 😅 

We can discuss more if needed, but basically I'd like tables that:

- I can set custom widths for columns
- Align text at the top of cells, whether it's a paragraph or list. Lists are really only an issue when they're the first thing in a cell, like [here](https://docs.dagster.io/dagster-plus/managing-deployments/branch-deployments#setting-up-branch-deployments)

File: docs/markdoc-component-documentation.md [hunk 4]
```diff
@@ -0,0 +1,263 @@
...
+
+`Inline code`
+
+```
+Code fences
+```
+
+Markdoc doesn't have syntax highlighting out of the box. We're going to need to add that in. When it get's added in, this python code should be beautifully highlighted.
+
+```python
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:57]: The one thing that's stopped me from migrating content to Markdoc is that we're not currently able to pull in code snippets from the examples folder (`/examples`).

The way we currently do this has two steps:

1. Use a `file` parameter after the language specified on a code fence, ex:
   ````
   ```python file=/docs_snippets/docs_snippets/concepts/assets/asset_group.py
   ```
   ````
2. Run `make mdx-format`, which injects the contents of the specified file into the code block. In this example, we'd then see: 
   ````
   ```python file=/docs_snippets/docs_snippets/concepts/assets/asset_group.py
   from dagster import asset


   @asset
   def upstream_asset():
       return [1, 2, 3]


   @asset
   def downstream_asset(upstream_asset):
       return upstream_asset + [4]
   ```
   ````

I'm not sure of the right/easy way to handle this in Markdoc, but if I had this, I could start migrating everything over and using these fabulous components. Perhaps discuss and then a follow-up PR to this one?

File: docs/markdoc-component-documentation.md [hunk 5]
```diff
@@ -0,0 +1,263 @@
...
+
+if __name__ == "__main__":
+    magic_8_ball()
+
+```
+
+# Custom Markdoc Tags
+We've extended markdown with custom tags that let us make the docs richer and more interactive.
+
+## Admonitions
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:100]: Could we take this opportunity to update some of the styling on these? Specifically, aligning the icon and copy to the top - this has always driven me nuts.

Lmk if that should be considered scope creep for this PR.

File: docs/markdoc-component-documentation.md [hunk 6]
```diff
@@ -0,0 +1,263 @@
...
+` {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} `
+
+{% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %}
+
+### Button Container
+The main use case I've seen for buttons in the docs is basically setting up multiple styling links in a row. Doing this requires putting buttons into a `ButtonContainer`
+
+{% buttonContainer %} {% button link="https://dog.ceo/" style="primary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="secondary" %} Click Me! {% /button %} {% button link="https://dog.ceo/" style="blurple" %} Click Me! {% /button %} {% /buttonContainer %}
+
+## Crosses and Checks
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:146]: I love these so much

File: docs/markdoc-component-documentation.md [hunk 7]
```diff
@@ -0,0 +1,263 @@
...
+
+You can also put crosses and checks into headers 
+
+### This is a header with a check {% check /%}
+Which is pretty neat.
+
+### This is a header with a cross {% cross /%}
+Which is also pretty neat.
+
+## Images
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:162]: When using this, do we have to specify the dimensions? Currently, if we use Markdown images and then run `make mdx-format`, the script will convert the images to `<Image>` tags and add the dimensions to the tag for us. It'd be nice to not have to do this manually - I can see myself and others forgetting to do it.

File: docs/markdoc-component-documentation.md [hunk 8]
```diff
@@ -0,0 +1,263 @@
...
+`{% image src="/images/some-image.png" width=300 height=200 alt="Text for screenreaders. Usually you want to describe things here." /%}`
+
+<!-- {% image src="/images/concepts/assets/asset-activity-observation.png" width=300 height=200 alt="Text for screenreaders. Usually you want to describe things here." /%} -->
+
+{% image src="/images/concepts/assets/asset-activity-observation.png" width=1758 height=1146 alt="Text go here" /%}
+
+
+## Badges
+
+`{% badge text="Badgey the Badger" /%}` lets you put a custom badge onto the page like this. {% badge text="Badgey the Badger" /%}
```
Comment (erinkcochran87) [inline docs/markdoc-component-documentation.md:174]: 🦡

**Review by erinkcochran87 — COMMENTED** [review]
You're doing the lord's work here 🙏 I left a few comments / questions, but overall I love this!

File: docs/markdoc-component-documentation.md (hunk 4 above)
Comment (nikomancy) [reply docs/markdoc-component-documentation.md:57]: I think we should discuss. I want to understand more about why we are doing code samples this way because there might be an easier approach.

File: docs/markdoc-component-documentation.md (hunk 5 above)
Comment (nikomancy) [reply docs/markdoc-component-documentation.md]: I can definitely get that put together. If you have any other styling tweaks you're looking for, we can break them out into a separate story. Otherwise, I'll try to get something together for this tomorrow morning.

File: docs/markdoc-component-documentation.md (hunk 2 above)
Comment (nikomancy) [reply docs/markdoc-component-documentation.md:193]: I'm sure there is something I can do to address this, and I would love to try. However, it is more economical to just custom-set the anchor links.

File: docs/markdoc-component-documentation.md (hunk 7 above)
Comment (nikomancy) [reply docs/markdoc-component-documentation.md:199]: I did not know that script existed. That is a much more convenient way to approach that. Let me research how it works and see if I can't either extend it to work on Markdoc or make a new script for the Markdoc tags.

File: docs/markdoc-component-documentation.md (hunk 2 above)
Comment (erinkcochran87) [reply docs/markdoc-component-documentation.md:193]: Sounds good to me!

File: docs/markdoc-component-documentation.md (hunk 7 above)
Comment (erinkcochran87) [reply docs/markdoc-component-documentation.md:199]: 🙏

**Review by erinkcochran87 — APPROVED** [review]
This looks great! I'm so excited to get to use these.
